source/module_hsolver/diago_iter_assist.h

```diff
@@ -49,7 +49,9 @@ class DiagoIterAssist
         int psi_nr,
         int psi_nc,
         psi::Psi<T, Device> &evc,
-        Real *en);
+        Real* en,
+        const std::function<void(T*, const int)>& add_to_hcc = [](T* null, const int n) {},
```

Comment:
do not add new functions in diago_iter_assist

Reply:
I've discussed with @haozhihan. These two functions does not induce any new dependence and is not conflict to his refactor.